remoting/src/main/java/org/apache/rocketmq/remoting/protocol/RocketMQSerializable.java

```diff
@@ -157,6 +162,9 @@ public static RemotingCommand rocketMQProtocolDecode(final byte[] headerArray) {
         // HashMap<String, String> extFields
         int extFieldsLength = headerBuffer.getInt();
         if (extFieldsLength > 0) {
+            if (extFieldsLength > headerArray.length) {
```

Comment:
Would it be more appropriate to replace `headerArray.length` with `headerBuffer.capacity() - headerBuffer.position()`?

Reply:
> Would it be more appropriate to replace `headerArray.length` with `headerBuffer.capacity() - headerBuffer.position()`?

`headerBuffer.capacity() - headerBuffer.position()` is more precise than `headerArray.length`.
Howerver, `headerBuffer.capacity() - headerBuffer.position()` has more subtraction operation than `headerArray.length`. Considering performance, `headerArray.length` may be more appropriate.